platform-includes/configuration/before-send-hint/native.mdx

```diff
@@ -17,7 +17,7 @@ sentry_value_t before_send(sentry_value_t event, void *hint) {
   const char *type_str = sentry_value_as_string(type);
 
   /* use the data passed during initialization */
-  const char *custom_error = (const char *)hint;
+  const char *custom_error = (const char *)user_data;
```

Comment:
I think this is still confusing, because characterizes `user_data` as something that should contribute to error resolution, which is absolutely not what it is used for. `user_data` is about passing in context from the user, so they can call into their code or retrieve relevant data.